cypress/e2e/buy-shirt.cy.ts

```diff
@@ -1,22 +1,41 @@
+import {
+  AddressStepPage,
+  LoginPage,
+  MenuContentPage,
+  PaymentStepPage,
+  ProductsListPage,
+  ShippingStepPage,
+  ShopingCartPage,
+} from "../page/index";
+
+const menuContentPage = new MenuContentPage();
+const productsListPage = new ProductsListPage();
+const shopingCartPage = new ShopingCartPage();
+const loginPage = new LoginPage();
+const addressStepPage = new AddressStepPage();
+const shippingStepPage = new ShippingStepPage();
+const paymentStepPage = new PaymentStepPage();
+
 describe("Buy a t-shirt", () => {
   it("then the t-shirt should be bought", () => {
-    cy.visit("http://automationpractice.com/");
-    cy.get("#block_top_menu > ul > li:nth-child(3) > a").click();
-    cy.get("#center_column a.button.ajax_add_to_cart_button.btn.btn-default").click();
-    cy.get("[style*='display: block;'] .button-container > a").click();
-    cy.get(".cart_navigation span").click();
+    menuContentPage.visitMenuContentPage();
+    menuContentPage.goToTShirtMenu();
+
+    productsListPage.addTShirtToCart();
+
+    shopingCartPage.goToCartPage();
+    shopingCartPage.goToCheckoutPage();
 
-    cy.get("#email").type("aperdomobo@gmail.com");
-    cy.get("#passwd").type("WorkshopProtractor");
+    loginPage.fillLoginForm("aperdomobo@gmail.com", "WorkshopProtractor");
+    loginPage.login();
 
-    cy.get("#SubmitLogin > span").click();
-    cy.get(".cart_navigation > .button > span").click();
+    addressStepPage.goToShippingStep();
 
-    cy.get("#cgv").check();
-    cy.get(".cart_navigation > .button > span").click();
+    shippingStepPage.acceptTermsAndConditions();
+    shippingStepPage.goToPaymentStep();
 
-    cy.get(".bankwire").click();
-    cy.get("#cart_navigation > .button > span").click();
+    paymentStepPage.selectPaymentMethod();
+    paymentStepPage.confirmOrder();
 
     cy.get("#center_column > div > p > strong")
         .should("have.text", "Your order on My Store is complete.");
```

Comment:
Why did you leave this part without POM?

Use the POM to retrieve the HTML component and assert it here